language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/Instances.hs

```diff
@@ -82,33 +76,36 @@ instance KnownType Integer where
     readKnown eval term = do
         -- 'term' is supposed to be already evaluated, but calling 'eval' is the easiest way
         -- to turn 'Error' into 'EvaluationFailure', which we later 'lift' to 'Convert'.
-        res <- makeRightReflectT $ eval mempty term
+        res <- eval mempty term
         case res of
             Constant () (BuiltinInt () i) -> pure i
-            _                             -> throwError "Not a builtin Integer"
+            _                             ->
+                throwingWithCause _UnliftingError "Not a builtin Integer" $ Just term
```

Comment:
Putting terms here as well as in other evaluation errors (i.e. putting terms everywhere) makes it much easier to use prismatic error handling (and generally makes sense).